app/uk/gov/hmrc/gform/sharedmodel/formtemplate/BooleanExpr.scala

```diff
@@ -31,17 +37,96 @@ final case object IsFalse extends BooleanExpr
+        lessThanOrEquals(prepopData(field1, retrievals, data), prepopData(field2, retrievals, data))
+      case Or(expr1, expr2)  => isTrue(expr1, data, retrievals) | isTrue(expr2, data, retrievals)
+      case And(expr1, expr2) => isTrue(expr1, data, retrievals) & isTrue(expr2, data, retrievals)
+      case IsTrue            => true
+      case _                 => false
+    }
+  }
+
+  private def equals(left: String, right: String): Boolean =
+    (toMaybeBigDecimal(left), toMaybeBigDecimal(right), left, right) match {
+      case (Some(l), Some(r), _, _) => (l == r)
+      case (_, _, l, r)             => (l == r)
+    }
+
+  private def greaterThan(left: String, right: String): Boolean =
+    (toMaybeBigDecimal(left), toMaybeBigDecimal(right), left, right) match {
+      case (Some(l), Some(r), _, _) => (l > r)
+      case (_, _, l, r)             => (l > r)
+    }
+
+  private def greaterThanOrEquals(left: String, right: String): Boolean =
+    (toMaybeBigDecimal(left), toMaybeBigDecimal(right), left, right) match {
+      case (Some(l), Some(r), _, _) => (l >= r)
+      case (_, _, l, r)             => (l >= r)
+    }
+
+  private def lessThan(left: String, right: String): Boolean =
+    (toMaybeBigDecimal(left), toMaybeBigDecimal(right), left, right) match {
+      case (Some(l), Some(r), _, _) => (l < r)
+      case (_, _, l, r)             => (l < r)
+    }
+
+  private def lessThanOrEquals(left: String, right: String): Boolean =
+    (toMaybeBigDecimal(left), toMaybeBigDecimal(right), left, right) match {
+      case (Some(l), Some(r), _, _) => (l <= r)
+      case (_, _, l, r)             => (l <= r)
     }
+
+  def prepopData(expr: Expr, retrievals: Retrievals, data: Map[FormComponentId, Seq[String]]): String = {
+
+    def toBigDecimal(str: String): BigDecimal =
+      Try(BigDecimal(str.replace(",", ""))) match {
+        case Success(x) => x
+        case Failure(_) => BigDecimal(0)
+      }
+
+    expr match {
+      case Constant(value) => value
+      case UserCtx(_)      => retrievals.affinityGroupName
+
+      case Add(field1, field2) => {
+        val y = prepopData(field1, retrievals, data)
+        val z = prepopData(field2, retrievals, data)
+        (toBigDecimal(y) + toBigDecimal(z)).toString()
+      }
+
+      case Subtraction(field1, field2) =>
+        val y = prepopData(field1, retrievals, data)
+        val z = prepopData(field2, retrievals, data)
+        (toBigDecimal(y) - toBigDecimal(z)).toString()
+
+      case Multiply(field1, field2) =>
+        val y = prepopData(field1, retrievals, data)
+        val z = prepopData(field2, retrievals, data)
+        (toBigDecimal(y) * toBigDecimal(z)).toString()
+
+      case id: FormCtx => data.get(id.toFieldId).map(_.head).getOrElse("")
+      case _           => ""
+    }
+  }
+
+  private def toMaybeBigDecimal(str: String): Option[BigDecimal] =
+    Try(BigDecimal(str.replace(",", ""))) match {
+      case Success(x) => Some(x)
+      case Failure(_) => None
+    }
```

Comment:
`match` on `Try` may be [replaced by `toOption` ](https://github.com/scala/scala/blob/2.11.x/src/library/scala/util/Try.scala#L159)